packages/core/src/types.ts

```diff
@@ -55,7 +55,7 @@ type GriffelStylesCSSObjectCustomL5 = {
 } & Partial<GriffelStylesUnsupportedCSSProperties> &
   GriffelStylesStrictCSSObject;
 
-export type GriffelStyle = GriffelStylesCSSObjectCustomL1;
+export type GriffelStyle = GriffelStylesStrictCSSObject | GriffelStylesCSSObjectCustomL1;
```

Comment:
I originally wanted to remove the strict object here. Unfortunately, without that object spreading (mixins) does not pass the type check in TS<4.2:

```ts
type T1 = {
  [Property: string]: string | number
}

type T2 = {
  n: number
}

type Compound = T1 & Partial<T2>

function assertType(t: Compound): Compound {
  return t;
}

const mixin: Compound = {
  color: 'red'
}

assertType({...mixin, background: 'blue'}) // 🔴 Type '{ background: string; n?: number | undefined; }' is not assignable to type 'T1' in TS<4.2
```
[Typescript playground](https://www.typescriptlang.org/vo/play?ts=3.9.7#code/C4TwDgpgBAKgjFAvFA3gWAFBSgbQAoBOA9pAaAFxQDOwBAlgHYDmAupTfc1AD5QMCuAWwBGEApgC+mTKEiwATElSZsDSgJFjJ0jLOgBhIoLBF+DACZL4UAGRQ8AQzJ0HAGwA8MeQD4dAMzMAY2A6IgYoByoqMWAYcAgACmBKQ2NTCwBKFKMTM0t0LCgCCGB+AnDgAG5tDExAsJooQToAD0ZstLylAux61yICSgByYvMhmsxI6LI4yASUADol5raGABooYQdAgGsmYjzh4Vd+CCGJDKgAeiuoQF4NwBZd2HioIZRN7b2Di3ZaRiZKnwAPzqISiAg8KB5CB+RgQcyAiRDKB0Kh8IjACJROhMBgOY7QYBEKB6V7wZGMWAAZXcABYFvIgA)